### ISSUE TITLE
CODEBOX_API_KEY=local generate a Codebox error expecting an OPENAI_API_KEY? in jupyter_runtime

### ISSUE THREAD

**Issue by richtong** [snakajima/SlashGPT#286]
Is local not supported it says this works in the documentation to use codebox, but not clear on why. If I don't set it ipython works fine. What is needed to run Codebox, the OPENAI_API_KEY is set? Looking at the code in the jupyter runtime function it looks like this is in codebox.start I see in line 46 is doesn't set the key if it is set to local, but then I think line 70, it callset codbox.start and the key isn't set if it is local.

**Comment by snakajima** [comment]
You need to have OPENAI_API_KEY regardless. CODEBOX_API_KEY changes the behavior of code execution (as the result of function call), not LLM.

**Comment by richtong** [comment]
Yes it’s strange the openai key works for regular calls. Looks like it’s deep in the code box call. Didn’t have time to read it. The key is in the environment but isn’t in a python variable. Switched to Ipython and except for the currency call it works.

I’m in the middle of adding some switch logic for fun (build an llm branching system and a scheduler :-) do tomorrow is understanding chat histories. Finding the manifest.Md and call architecture helped a lot. I linked to it in my private readme…

Sent via Superhuman iOS ( ***@***.*** )
